### DISCUSSION TITLE
Recommended way to add linting in tamagui/expo monorepo

### DISCUSSION THREAD

**Discussion by matthewhausman** [tamagui/tamagui#732] [Q&A]
In the create-tamagui starter monorepo, for example, how would one get eslint support. I'm scared that one day I'll forget a key on a component when mapping in jsx and be completely screwed.

**Comment by matthewhausman** [comment]
Settled upon something that works pretty damn well, but it would be super nice if there was a tamagui-specific plugin that would alter the  **react-native/no-raw-text** rule so that text could be allowed in buttons, paragraphs, headings, selects, etc...

my `.eslint.json` file in the root of the project's directory

```
{
    "env": {
        "browser": true,
        "es2021": true,
        "react-native/react-native": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended",
        "plugin:react-native/all",
        "plugin:@typescript-eslint/recommended"
    ],
    "overrides": [
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "plugins": [
        "react",
        "react-native",
        "@typescript-eslint"
    ],
    "rules": {
        "react/react-in-jsx-scope": "off",
        "react-native/no-inline-styles": "off",
        "react-native/no-color-literals": "off",
        "react-native/no-raw-text": "off"
    }
}
```

**Reply by matthewhausman** [reply]
and ofc 

`yarn add -D eslint-plugin-react eslint-plugin-react-native`

**Reply by foufrix** [reply]
Thank you for the starter config file for tamagui, helps a lot.
you can also add `yarn add -D @typescript-eslint/eslint-plugin @typescript-eslint/parser` and `yarn add eslint`

**Comment by matthewmturner** [comment]
@matthewhausman did you only need to configure eslint at the root level? nothing in packages or apps?
